### PR TITLE
Add a small generator example script

### PR DESCRIPTION
**Pull request by danieldk**
## Description

This example reads prompts as a JSON list and returns JSON list with the generated text.

### Types of change

Example

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: examples/generate.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+parser.add_argument(
+    "--sample",
+    "-s",
+    action="store_true",
+    help="enable sampling",
+)
+parser.add_argument(
+    "--torch-sdp",
+    action="store_true",
+    help="enable Torch scaled dot product implementation",
```
Comment (shadeMe) [inline examples/generate.py:59]:
```suggestion
    help="enable Torch scaled dot product attention implementation",
```

File: examples/generate.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+from curated_transformers.generation.config import (
+    GreedyGeneratorConfig,
+    SampleGeneratorConfig,
+)
+from curated_transformers.models.attention import enable_torch_sdp
+from curated_transformers.util.auto_model import AutoGenerator
+
+parser = argparse.ArgumentParser(
+    description="Generate from a list of prompts in JSON format",
```
Comment (shadeMe) [inline examples/generate.py:18]: It'd be good to also include an example of the format.

Comment (danieldk) [reply examples/generate.py:18]: But where? 🤔

Comment (shadeMe) [reply examples/generate.py:18]: I was thinking we could add it to the description string? Doesn't it get shown in the help text?

Comment (danieldk) [reply examples/generate.py:27]: Added an epilogue.

**Review by shadeMe — APPROVED** [review]

**Pull request merged**